packages/kit/src/components/ComponentForm/ComponentForm.component.js

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import { cmfConnect } from '@talend/react-cmf';
+import { UIForm } from '@talend/react-forms/lib/UIForm';
+import omit from 'lodash/omit';
+import { Map } from 'immutable';
+import { CircularProgress } from '@talend/react-components';
+import kit from 'component-kit.js';
+
+export const DEFAULT_STATE = new Map({
+	dirty: false,
+});
+
+class ComponentForm extends React.Component {
+	static displayName = 'ComponentForm';
+	static propTypes = {
+		...cmfConnect.propTypes,
+	};
+	static defaultProps = {
+		customTriggers: () => {},
+	};
+
+	static ON_CHANGE = 'TCOMP_FORM_CHANGE';
+	static ON_TRIGGER = 'TCOMP_FORM_TRIGGER';
+	static ON_SUBMIT = 'TCOMP_FORM_SUBMIT';
+	static ON_DEFINITION_URL_CHANGED = 'TCOMP_FORM_DEFINITION_URL_CHANGE';
+
+	constructor(props) {
+		super(props);
+		this.state = {};
+		this.onTrigger = this.onTrigger.bind(this);
+		this.onChange = this.onChange.bind(this);
+		this.onSubmit = this.onSubmit.bind(this);
+		this.getUISpec = this.getUISpec.bind(this);
+		this.setupTrigger = this.setupTrigger.bind(this);
+		this.getTriggers = this.getTriggers.bind(this);
+		this.setupTrigger(props);
+	}
+
+	componentDidUpdate(props) {
+		if (
+			props.triggerURL !== this.props.triggerURL ||
+			props.customTriggers !== this.props.customTriggers
+		) {
+			this.trigger = this.setupTrigger(this.props);
+		}
+		if (this.props.definitionURL !== props.definitionURL) {
+			this.props.dispatch({
+				type: ComponentForm.ON_DEFINITION_URL_CHANGED,
+				...this.props,
+			});
+		}
+	}
+
+	onChange(event, data) {
+		event.persist();
+		if (!this.props.state.get('dirty')) {
```

Comment:
You don't need this `if`, just set `dirty: true` when you set the properties in state